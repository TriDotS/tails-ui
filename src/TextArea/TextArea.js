import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const StyledTextArea = styled.textarea`
  &:focus {
    outline: none;
  }
`

const TextArea = ({ label, value, type, name, placeholder, color, rows }) => (
  <div>
    {label && (
      <label className="block uppercase tracking-wide text-grey-darker text-xs font-bold mb-2">
        {label}
      </label>
    )}
    <StyledTextArea
      className={`bg-grey-lighter appearance-none border-2 border-grey-lighter hover:border-${color}  rounded w-full py-2 px-4 text-grey-darker`}
      type={type}
      value={value}
      name={name}
      placeholder={placeholder}
      rows={rows}
    />
  </div>
)

TextArea.defaultProps = {
  rows: 4,
  color: 'teal'
}

TextArea.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  color: PropTypes.string,
  rows: PropTypes.number
}

export default TextArea
