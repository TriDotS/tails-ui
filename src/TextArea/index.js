import React from 'react'
import styled from 'styled-components'

const TextArea = styled.textarea`
  padding: 10px 10px;
  box-sizing: border-box;
  border: 1px solid silver;
  font-size: 1em;
  border-radius: ${props => (props.rounded ? '3px' : '0px')};
  transition: 0.5s;
  outline: none;

  &:focus {
    border: 1px solid dimgray;
  }

  &:invalid {
    border: 1px solid indianred;
  }
`

const FieldContainer = styled.div`
  display: inline-flex;
  flex-direction: column;
  width: ${props => (props.fullWidth ? '100%' : 'initial')};
`

export default ({ placeholder, labelText, fullWidth, rounded }) =>
  <FieldContainer fullWidth={fullWidth}>
    {labelText && <label>{labelText}</label>}
    <TextArea
      placeholder={placeholder || null}
      rounded={rounded}
      labelText={labelText}
    />
  </FieldContainer>
