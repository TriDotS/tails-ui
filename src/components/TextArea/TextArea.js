import PropTypes from 'prop-types'
import React from 'react'

const TextArea = ({
  label,
  value,
  type,
  name,
  placeholder,
  color,
  rows,
  hasError,
  ...props
}) => (
  <div>
    {label && (
      <label className="block uppercase tracking-wide text-grey-darker text-xs font-bold mb-2">
        {label}
      </label>
    )}
    <textarea
      className={`bg-grey-lighter appearance-none border-2 ${
        hasError ? 'border-red' : 'border-grey-lighter'
      } ${
        hasError ? 'hover:border-red' : `hover:border-${color}`
      }  rounded w-full py-2 px-4 text-grey-darker`}
      type={type}
      value={value}
      name={name}
      placeholder={placeholder}
      rows={rows}
      {...props}
    />
  </div>
)

TextArea.defaultProps = {
  rows: 4,
  color: 'teal'
}

TextArea.propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  type: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  color: PropTypes.string,
  rows: PropTypes.number,
  hasError: PropTypes.bool
}

export default TextArea
