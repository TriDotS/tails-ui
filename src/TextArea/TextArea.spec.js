import React from 'react'
import TextArea from './TextArea'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<TextArea />, div)
})

it('generate text area snapshot', () => {
  const component = renderer.create(<TextArea />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
