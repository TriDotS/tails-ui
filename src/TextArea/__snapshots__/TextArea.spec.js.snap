// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate text area snapshot 1`] = `
<div
  className="sc-bwzfXH jqtxFq"
>
  <textarea
    className="sc-bdVaJa glEYay"
    placeholder={null}
  />
</div>
`;
